modules/ting/ting.entities.inc

```diff
@@ -485,22 +448,32 @@ class TingCollection extends DingEntityCollection {
     return $this->types = array_unique($types);
   }
 
+  /**
+   * Gets the number of different types of materials in the collection.
+   *
+   * @return array
+   *   Count of the various types, keyed by type name.
+   */
   function getTypes_count() {
     $types = array();
     foreach ($this->entities as $object) {
-      if ( !isset($types[$object->type] ) ) {
+      if (!isset($types[$object->type])) {
         $types[$object->type] = 1;
-      } else {
+      }
+      else {
         $types[$object->type]++;
       }
     }
     return $this->types_count = $types;
   }
 
+  /**
+   * Get the primary Object in this collection.
+   *
+   * @return \Ting\TingObjectInterface|NULL
```

Comment:
Allowing `NULL` as a return value here seems odd. I would think a collection would always be initiated based off one or more objects. If that is the case there will always be a primary object.

Reply:
I'll update the code and comment